webui/react/src/pages/ExperimentDetails/TrialsComparisonModal.tsx

```diff
@@ -155,13 +156,15 @@ export const TrialsComparisonTable: React.FC<TableProps> = ({
   );
 
   const loadableMetrics = useMetricNames(experimentIds, handleMetricNamesError);
-  const metrics = Loadable.getOrElse([], loadableMetrics);
+  const metrics: Metric[] = useMemo(() => {
+    return Loadable.getOrElse([], loadableMetrics);
+  }, [loadableMetrics]);
 
   const prevMetrics = usePrevious(metrics, []);
 
   useEffect(() => {
     setSelectedMetrics((prevSelectedMetrics) =>
-      isEqual(prevSelectedMetrics, prevMetrics) ? metrics : prevSelectedMetrics,
+      _.isEqual(prevSelectedMetrics, prevMetrics) ? metrics : prevSelectedMetrics,
```

Comment:
nit: if we want to use lodash behind the scenes in the implementation of our `isEqual` util that's fine, but I don't think we should call it directly here unless we have to

Reply:
its a partial change for https://hpe-aiatscale.atlassian.net/browse/WEB-1086, but i'll revert it